gdk/wizard/commands/prompt.py

```diff
@@ -0,0 +1,207 @@
+        ----------
+        field_map : data(dict)
+            A dictionary object containing the configuration from greengrass project config file.
+
+        """
+
+        self.data = Wizard_data()
+        self.getter = Wizard_getter(self.data)
+        self.setter = Wizard_setter(self.data)
+        self.checker = Wizard_checker(self.data)
+
+        self.parser = argparse.ArgumentParser()
+
+    def prompter(self, field, value, required, max_attempts=3):
+        """
+        Parameters
+        ----------
+            field (string): a field key of the gdk-config file to be changed
+            value (string): the current value corresponding the field key to be changed
+            parser (ArgumentParser): parser for retriving command line arguments
+
+        Returns
+        -------
+            string: the value for the field key "field"
+        """
+        require = "required " if required else "optional "
+        for attempt in range(1, max_attempts + 1):
+            # args = self.parser.parse_args(
+            #     [f'--{field}', input(f"Current value of the {require}{field} is {value}: ")]
+            # )
+            args = self.parser.parse_args(
+                [f"--{field}", self.interactive_prompt(field, value, require)]
+            )
+
+            response = getattr(args, field)
+            if self.checker.is_valid_input(response, field):
+                return response
+            print(
+                f"Attempt {attempt}/{max_attempts}: Invalid response. Please try again."
+            )
+
+        print("Exceeded maximum attempts. Assuming default response.")
+        return value
+
+    def add_parsers_arguments(self):
+        for field in Fields:
+            self.parser.add_argument(f"--{field.value}")
+
+    def change_build_or_publish(self, build_or_publish, max_attempts=3):
+        """
+        Prompts the users to answer if they would like to change the field value
+        of a particular field in the gdk-config file
+
+        Parameters
+        ----------
+            field (string): a field key of the gdk-config file to be changed
+            value (string): the current value corresponding the field key to be changed
+            parser (ArgumentParser): parser for retriving command line arguments
+
+        Returns
+        -------
+            boolean: True if the user answers 'y' they do want to change the value of field 'field'
+                    and False if the user answers 'n' they do not want to change the value of that field
+
+        """
+        self.parser.add_argument(
+            f"--change_{build_or_publish}",
+            help=f"Change componenet {build_or_publish} configurations",
+        )
+
+        for _ in range(max_attempts):
+            args = self.parser.parse_args(
+                [
+                    f"--change_{build_or_publish}",
+                    input(
+                        f"Do you want to change the {build_or_publish} configurations? (y/n) "
```

Comment:
Why do you want input from Customer? Even if they select "n" the fields will still be editable..right? Or you would simply exit the wizard? What is the behavior in each scenario?

Reply:
The idea was to allow users to be prompted in an hierarchical manner. 'custom_build_command' and 'build_system' fall under the umbrella of build and "bucket", "region", and "options" all fall under the umbrella of publish, so if we first ask the user if they would like to change the build config or regions at all before prompting to change specific fields under their respective umbrellas. If the user respond 'n' the prompter would simply skip over the fields in build or publish.